### PR TITLE
update promocodes to use config values in migration

### LINKED ISSUE #71: apply promocode facade does not use the column name specified in config file
**Describe the bug**
If i change the  'foreign_pivot_key' => 'promocode_id' (default) to  'foreign_pivot_key' => 'promo_code_id', then apply promoccode gives the error "promocode_id column not found in pivot table"

**To Reproduce**
1. Install this package
2. Goto promocodes.php config file and set 'foreign_pivot_key' => 'promo_code_id'
3. Generate any promocode and try to apply with the authenticated user.
4. See error ("promocode_id column not found in pivot table")

please refer to the funccitonal below:

/**
     * Apply promocode to user that it's used from now.
     *
     * @param string $code
     *
     * @return bool|Promocode
     * @throws AlreadyUsedException
     * @throws UnauthenticatedException
     */
    public function apply($code)
    {
        if (!auth()->check()) {
            throw new UnauthenticatedException;
        }

        if ($promocode = $this->check($code)) {
            if ($this->isSecondUsageAttempt($promocode)) {
                throw new AlreadyUsedException;
            }

            $promocode->users()->attach(auth()->id(), [
                'promocode_id' => $promocode->id,
                'used_at' => Carbon::now(),
            ]);

            if (!is_null($promocode->quantity)) {
                $promocode->quantity -= 1;
                $promocode->save();
            }

            return $promocode->load('users');
        }

        return false;
    }

In the above function, promocode_id is hardcoded which should be readable from config file.

### PR DESCRIPTION
**Pull request by neolikotsi**
closes #75 closes #71

### CODE REVIEW

File: tests/CustomMigrationConfigurationTest.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        $this->app['config']->set('promocodes.user_model', User::class);
+
+        $this->app['config']->set('promocodes.table', 'coupons');
+        $this->app['config']->set('promocodes.relation_table', 'coupon_user');
+        $this->app['config']->set('promocodes.foreign_pivot_key', 'coupon_id');
+        $this->app['config']->set('promocodes.related_pivot_key', 'user_id');
+    }
+
+    /** @test */
+    public function it_migrates_with_custom_config_values()
```
[the remark this answers is not in the document]
Comment (neolikotsi) [reply tests/CustomMigrationConfigurationTest.php:32]: use snake case like previous tests.

File: tests/CustomMigrationConfigurationTest.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    {
+        $this->assertTrue(Schema::hasTable('coupons'));
+        $this->assertTrue(Schema::hasTable('coupon_user'));
+        $this->assertTrue(Schema::hasColumns('coupon_user', ['coupon_id', 'user_id']));
+    }
+
+    /** @test */
+    public function it_can_create_code_with_custom_migration_config_values_in_database()
```
[the remark this answers is not in the document]
Comment (neolikotsi) [reply tests/CustomMigrationConfigurationTest.php:40]: use snake case like previous tests.

File: tests/CustomMigrationConfigurationTest.php [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    {
+        Promocodes::create();
+
+        $this->assertCount(1, Promocodes::all());
+    }
+
+    /** @test */
+    public function it_can_reedem_code_with_custom_migration_config_values()
```
[the remark this answers is not in the document]
Comment (neolikotsi) [reply tests/CustomMigrationConfigurationTest.php:48]: use snake case like previous tests.

**Comment by zgabievi** [comment]
Thank you for your effort. This is already implemented in new release.

**Pull request closed without merging**